satpy/readers/core/utils.py

```diff
@@ -343,21 +344,18 @@ def unzip_context(filename):
 
 @contextmanager
 def generic_open(filename, *args, **kwargs):
-    """Context manager for opening either a regular file or a bzip2 file.
+    """Context manager for opening either a regular file or a compressed file.
 
-    Returns a file-like object.
+    Yields an open file-like object.
     """
-    if os.fspath(filename).endswith(".bz2"):
-        fp = bz2.open(filename, *args, **kwargs)
-    else:
-        try:
-            fp = filename.open(*args, **kwargs)
-        except AttributeError:
-            fp = open(filename, *args, **kwargs)
     try:
+        path = filename.to_upath()
+    except AttributeError:
+        path = UPath(filename)
```

Comment:
```suggestion
        # satpy.readers.core.remote.FSFile instance
        path = filename.to_upath()
    except AttributeError:
        # str or Path-like
        path = UPath(filename)
```

Are these comments correct? Can we add them?